integration/inspektor-gadget/run_trace_oomkill.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/run/types"
+
+	. "github.com/inspektor-gadget/inspektor-gadget/integration"
+)
+
+func runTraceOOMKill(t *testing.T, ns string, cmd string) {
+	// TODO: Handle it once we support getting container image name from docker
+	isDockerRuntime := IsDockerRuntime(t)
+
+	traceOOMKillCmd := &Command{
+		Name:         "StartRunTraceOOMKillGadget",
+		Cmd:          cmd,
+		StartAndStop: true,
+		ValidateOutput: func(t *testing.T, output string) {
+			expectedBaseJsonObj := RunEventToObj(t, &types.Event{
+				CommonData: BuildCommonData(ns, WithContainerImageName("docker.io/library/busybox:latest", isDockerRuntime)),
+			})
+
+			expectedTraceOOMKillJsonObj := map[string]interface{}{
+				"fpid":      0,
+				"fuid":      1000,
+				"fgid":      2000,
+				"tpid":      0,
+				"pages":     0,
+				"mntns_id":  0,
+				"timestamp": "",
+				"fcomm":     "",
+				"tcomm":     "tail",
+			}
+
+			expectedJsonObj := MergeJsonObjs(t, expectedBaseJsonObj, expectedTraceOOMKillJsonObj)
+
+			normalize := func(m map[string]interface{}) {
+				SetEventK8sNode(m, "")
+
+				// TODO: Verify container runtime and container name
+				SetEventRuntimeName(m, "")
+				SetEventRuntimeContainerID(m, "")
+				SetEventRuntimeContainerName(m, "")
+
+				m["fpid"] = uint32(0)
+				m["tpid"] = uint32(0)
+				m["pages"] = uint32(0)
+				m["mntns_id"] = 0
+				m["timestamp"] = ""
+				m["fcomm"] = ""
+			}
+
+			ExpectEntriesToMatchObj(t, output, normalize, expectedJsonObj)
+		},
+	}
+
+	limitPodYaml := fmt.Sprintf(`
```

Comment:
Do you think we could generalize https://github.com/inspektor-gadget/inspektor-gadget/blob/main/integration/command.go#L561 even more? So additional stuff we need (here `resources`) can be used?
I just don't like inline yaml :D

Reply:
What I meant is something like https://github.com/inspektor-gadget/inspektor-gadget/commit/9abf9d9bca4ee8885d3ac068ab57aa0d59d1227d where we keep the old `PodCommand` but add another function.